packages/teams-js/src/internal/utils.ts

```diff
@@ -429,6 +429,12 @@ export function validateUrl(url: URL, errorToThrow?: Error): void {
   }
 }
 
+export function fullyQualifyUrlString(urlString: string): URL {
```

Comment:
Interesting to notice: 
fullyQualifyUrlString is converting a string to Url type, then in validateUrl, it converts Url type back to string, because it needs to validate the tag and length etc on..

so wondering if it worths to update validateUrl method to accept a string, then after tag/length validation, we convert to url for https validation. i.e, it might have a chance to merge  fullyQualifyUrlString logics into validateUrl.

Reply:
It is interesting, but I think I want to leave this as-is. I would actually do a couple of "unnecessary" conversions within in internal functions in exchange for using the URL type as much as possible. There might be future improvements we can make there, but for now we have small, pretty understandable functions and have started to increase the number of URL objects we're using, which is good!